src/integrationTest/sam.test.ts

```diff
@@ -0,0 +1,181 @@
+/*!
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as assert from 'assert'
+import { mkdirpSync, readFileSync, removeSync } from 'fs-extra'
+import * as path from 'path'
+import * as vscode from 'vscode'
+import { SamLambdaRuntime } from '../../src/lambda/models/samLambdaRuntime'
+import { getSamCliContext } from '../../src/shared/sam/cli/samCliContext'
+import { runSamCliInit, SamCliInitArgs } from '../../src/shared/sam/cli/samCliInit'
+import { activateExtension, sleep, TIMEOUT } from './integrationTestsUtilities'
+
+const projectFolder = `${__dirname}`
+let projectSDK = ''
+let projectPath = ''
+let debuggerType = ''
+let documentUri: vscode.Uri
+
+const runtimes = [
+    { name: 'nodejs10.x', path: 'testProject/hello-world/app.js', debuggerType: 'node2' },
+    { name: 'python2.7', path: 'testProject/hello_world/app.py', debuggerType: 'python' },
+    { name: 'python3.6', path: 'testProject/hello_world/app.py', debuggerType: 'python' },
+    { name: 'python3.7', path: 'testProject/hello_world/app.py', debuggerType: 'python' },
+    { name: 'dotnetcore2.1', path: 'testProject/src/HelloWorld/Function.cs', debuggerType: 'coreclr' }
+]
+
+async function openSamProject(): Promise<vscode.Uri> {
+    const documentPath = path.join(projectFolder, projectPath)
+    await vscode.workspace.openTextDocument(documentPath)
+
+    return vscode.Uri.file(documentPath)
+}
+
+async function getCodeLenses(): Promise<vscode.CodeLens[]> {
+    let codeLenses: vscode.CodeLens[] | undefined
+    while (true) {
+        try {
+            const codeLensesPromise: Thenable<vscode.CodeLens[] | undefined> =
+                vscode.commands.executeCommand('vscode.executeCodeLensProvider', documentUri)
+            codeLenses = await codeLensesPromise
+            if (!codeLenses) {
+                continue
+            }
+            // omnisharp spits out some undefined code lenses for some reason, we filter them because they are
+            // not shown to the user and do not affect how our extension is working
+            codeLenses = codeLenses!.filter(lens => lens !== undefined && lens.command !== undefined)
+            if (codeLenses!.length === 3) {
```

Comment:
Does this work if you're testing a file with more than one Lambda handler? If this should work in that scenario, you may want to move this to `codeLenses!.length > 0 && codeLenses!.length % 3 === 0`

Also, you may want to make this value a variable...I'm assuming this is because each Lambda handler has 3 codelenses. Definitely define that, and also, what if we want to test something in the future with more codelenses?

Reply:
These tests are explicitly based on testing with one lambda handler, if we add tests that handle different amounts of handlers or other cases like that we should probably put them in a new test group and have a stable project that we commit to the repo as test resources